sound/soc/sof/intel/hda-loader.c

```diff
@@ -470,3 +471,122 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	/* re-enable clock gating and power gating */
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
+
+static int hda_dsp_ext_man_get_cavs_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_cavs_config_data *config_data =
+		container_of(hdr, struct sof_ext_man_cavs_config_data, hdr);
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int i;
+
+	for (i = 0; i < SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM - 1; i++)
+		switch (config_data->elems[i].token) {
+		case SOF_EXT_MAN_CAVS_CONFIG_CAVS_LPRO:
+			hda->clk_config = config_data->elems[i].value;
```

Comment:
is my understanding correct that you get this elem only when the FW is configured for LPRO? WHat happens otherwise, your clk_config will still be INT_MIN?

Reply:
if clk_config is not read from ext manifest, clk_config is remained as INT_MIN. This means we don't have any idea about clock config. When it read, 0:HPRO, 1:LPRO.

Reply:
no thats not right. Look at the FW code https://github.com/thesofproject/sof/pull/2893/commits/07775daa3183a6ac0889ccbc3090d731505072f7#
line 19, the config item will be added only if LPRO is enabled.

Reply:
You're right. `the config item will be added only if LPRO is enabled`
Then we don't need to initialize clk_config to INT_MIN. By default, it will be 0 which mean HPRO.
Currently ext manifest is not part of sof firmware,  but we can plan to merge LPRO dependent code(i.e, ICL ICCMAX) and ext manifest enabling PR together. Then clk_config init value concern is covered.

Reply:
yes, that correct. no need to even make is s32, u32 would be fine